ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/Leaf.java

```diff
@@ -218,10 +224,47 @@ void optimize() {
      * @return A set of available network actions after this leaf.
      */
     Set<NetworkAction> bloom() {
-        return raoData.getAvailableNetworkActions()
-                .stream()
-                .filter(na -> !networkActions.contains(na))
-                .collect(Collectors.toSet());
+        SearchTreeRaoParameters searchTreeRaoParameters = raoParameters.getExtension(SearchTreeRaoParameters.class);
+        if (searchTreeRaoParameters.getSkipNetworkActionsFarFromMostLimitingElement()) {
+            List<Optional<Country>> worstCnecLocation = getMostLimitingElementLocation();
+            return raoData.getAvailableNetworkActions()
+                    .stream()
+                    .filter(na -> !networkActions.contains(na)
+                            && isNetworkActionCloseToLocations(na, worstCnecLocation, searchTreeRaoParameters.getMaxNumberOfBoundariesForSkippingNetworkActions()))
+                    .collect(Collectors.toSet());
+        } else {
+            return raoData.getAvailableNetworkActions()
+                    .stream()
+                    .filter(na -> !networkActions.contains(na))
+                    .collect(Collectors.toSet());
+        }
+    }
+
+    /**
+     * Says if a network action is close to a given set of countries, respecting the maximum number of boundaries
+     */
+    public boolean isNetworkActionCloseToLocations(NetworkAction networkAction, List<Optional<Country>> locations, int maxNumberOfBoundaries) {
```

Comment:
you could use the variableClass raoParameters instead of having the maxNumberOfBoundaries in argument here

Reply:
ok